### PR TITLE
type hints in Python API

### PR DESCRIPTION
**Pull request by lukasmittag**
Added type hints in implementation of kuksa client. If I'm not wrong the type hints already exist in python library. Would solve #493

### CODE REVIEW

File: kuksa-client/kuksa_client/__init__.py [hunk 1]
```diff
@@ -51,32 +51,32 @@ def checkConnection(self):
     def stop(self):
         self.backend.stop()
 
-    # Do authorization by passing a jwt token or a token file
-    def authorize(self, token=None, timeout=5):
-        return self.backend.authorize(token, timeout)
+    # Do authorization by passing a token file
+    def authorize(self, tokenfile: str = None, timeout=5):
```
Comment (erikbosch) [inline kuksa-client/kuksa_client/__init__.py:55]: This is maybe a good naming convention to gradually introduce also in other places - use `token` when we refer to the contents of a token, and `tokenfile` when the content is supposed to be a path

File: kuksa-client/kuksa_client/grpc/__init__.py [hunk 2]
```diff
@@ -858,6 +858,11 @@ def subscribe(self, *, entries: Iterable[SubscribeEntry], **rpc_kwargs) -> Itera
 
     @check_connected
     def authorize(self, *, token: str, **rpc_kwargs) -> str:
+        """
+        Parameters:
+            rpc_kwargs
```
Comment (erikbosch) [inline kuksa-client/kuksa_client/grpc/__init__.py:863]: Even if we would have a naming convention - as we not yet are consistent - would it be an idea here to mention in a comment that token shall be the actual token, not a file path.

File: kuksa-client/kuksa_client/grpc/aio.py [hunk 3]
```diff
@@ -318,6 +318,11 @@ async def subscribe(self, *,
 
     @check_connected_async
     async def authorize(self, token: str, **rpc_kwargs) -> str:
```
Comment (erikbosch) [inline kuksa-client/kuksa_client/grpc/aio.py:320]: Should we possibly when changing here also add a `*, `after `self` to make it identical to the same method in ` __init__.py` and similar to the other methods in this file?

File: kuksa-client/kuksa_client/grpc/__init__.py (hunk 2 above)
Comment (lukasmittag) [reply kuksa-client/kuksa_client/grpc/__init__.py:863]: Yes, good point

File: kuksa-client/kuksa_client/grpc/aio.py (hunk 3 above)
Comment (argerus) [reply kuksa-client/kuksa_client/grpc/aio.py:320]: My question would be _why_ `*` is included in those other function signatures...

Comment (erikbosch) [reply kuksa-client/kuksa_client/grpc/aio.py:320]: Fair point - as long as it is consistent I am happy.

**Comment by erikbosch** [comment]
@lukasmittag  - apart from fixing the conflicts, is there something more to do? I think it would be good to get this one merged and then create a new kuksa-client pre-release

**Pull request merged**